packages/headless/src/controllers/controller/headless-controller.ts

```diff
@@ -15,13 +15,16 @@ export interface Controller {
 export function buildController<T extends object>(
   engine: CoreEngine<T>
 ): Controller {
-  let prevState: string;
+  const prevState: Map<Symbol, string> = new Map();
```

Comment:
Just a thought, what if instead we kept a list/map of the listener? When the state is changed, rather than comparing states multiple time could we do it once? 

Exemple we have 5 subscribers with each their listener with this solution:
- compares each identical prevState and call each listener

If we kept a listener's map and 1 prevState, we could compare once and call all listeners.
It might not be doable, we have to manage the unsubscribes correctly too

Reply:
I can try to come up with something tomorrow.
We can change it so that we just have one anonymous "internal" function hooked to engine.subscribe. It should be doable and will save on comparison time.